Problem3.rb

```diff
@@ -0,0 +1,48 @@
+require 'rspec'
+
+def swap(array, i, j)
+
+  temp = array[i]
+  array[i] = array[j]
+  array[j] = temp
+
+end
+
+def contains_duplicates?(array)
+
+  min = 1
+
+  array.each_index do |i|
+
+    temp = array[i]
+    j = temp - min
+
+    while i != j do

```

Comment:
"Do not use 2 nesting for loops since the time complexity would be high. Try to find a solution by executing just one loop". Consider using just one loop, not matter if there is a very small one. :)